app/service/data_svc.py

```diff
@@ -389,15 +426,39 @@ def __init__(self, dao):
                                                           info['cleanup'].strip().encode(
                                                               'utf-8')).decode() if info.get(
                                                           'cleanup') else None,
-                                                      payload=info.get('payload'), parser=info.get('parser'))
+                                                      payload=info.get('payload'), parser=info.get('parser'),
+                                                      relationships=fact_relationships)
                 await self._delete_stale_abilities(ab)
 
-    async def _save_ability_extras(self, identifier, payload, parser):
+    async def _relationships_from_yaml(self, ability):
+        relationships = []
+        if ability.get('requires'):
+            for r in ability.get('requires'):
+                line = await self._cleanse_facts(r)
+                relationships.append(dict(property1=line[0], relationship=line[1], property2=line[2],
+                                          relationship_type='requires'))
+        if ability.get('consequence'):
```

Comment:
I'm confident these two blocks of code (if requires/if consequence) can be consolidated.

Reply:
secondarily, make sure you test this functionality out by editing abilities (and creating new ones) from the GUI/Chain mode - as well as from disk/YML files.